src/main/java/com/iota/iri/service/tipselection/impl/TailFinderImpl.java

```diff
@@ -27,6 +27,11 @@ public TailFinderImpl(Tangle tangle) {
     @Override
     public Optional<Hash> findTail(Hash hash) throws Exception {
         TransactionViewModel tx = TransactionViewModel.fromHash(tangle, hash);
+        return findTailFromTx(tx);
```

Comment:
it's probably worth changing in the other usages - but not part of the scope; open an issue?

Reply:
Since you commented and it is such an easy fix, I simply changed it in SpentAddressesServiceImpl